xbmc/pvr/epg/Epg.cpp

```diff
@@ -339,10 +339,13 @@ bool CPVREpg::Delete(const std::shared_ptr<CPVREpgDatabase>& database)
   }
 
   // delete own epg db entry
-  database->Delete(*this);
+  if (bCommit)
```

Comment:
IMO, this method never gets called with bCommit=true. So, the sync code path can be removed. As well as the whole implementation of CPVREpgDatabase::Delete(CPVREpg).